services-custom/dynamodb-enhanced/src/test/java/software/amazon/awssdk/enhanced/dynamodb/model/QueryEnhancedRequestTest.java

```diff
@@ -89,6 +90,48 @@ public void builder_maximal() {
         assertThat(builtObject.attributesToProject(), is(attributesToProject));
     }
 
+    @Test
+    public void test_nestedAttributesToProject(){
+
+        String[] attributesToProjectArray = {"one", "two"};
+        String additionalElement = "three";
+        QueryEnhancedRequest builtObject = QueryEnhancedRequest.builder()
+                .addNestedAttributesToProject(NestedAttributeName.create("foo", "bar"))
+                .attributesToProject(attributesToProjectArray)
+                .addAttributeToProject(additionalElement)
+                .build();
+        List<String> attributesToProject = Arrays.asList("foo.bar", "one", "two", "three");
+        assertThat(builtObject.attributesToProject(), is(attributesToProject));
+
+
+        QueryEnhancedRequest builtObjectOne = QueryEnhancedRequest.builder()
+                .attributesToProject(attributesToProjectArray)
+                .addAttributeToProject(additionalElement)
+                .addNestedAttributesToProject(NestedAttributeName.create("foo", "bar"))
+                .build();
+        List<String> attributesToProjectNestedLast = Arrays.asList( "one", "two", "three","foo.bar");
+        assertThat(builtObjectOne.attributesToProject(), is(attributesToProjectNestedLast));
+
+
+        QueryEnhancedRequest builtObjectTwo = QueryEnhancedRequest.builder()
+                .attributesToProject(attributesToProjectArray)
+                .addAttributeToProject(additionalElement)
+                .addNestedAttributesToProject(NestedAttributeName.create("foo", "bar"))
+                .attributesToProject(additionalElement)
+                .build();
+        List<String> attributesToProjectStringAttributeOverwrittenNestedLast = Arrays.asList( "foo.bar", "three");
```

Comment:
See my other comment in the code, I think the 'correct' state here should be "three", and the nested attribute should be overwritten with the others.